src/virtual_dom/vtag.rs

```diff
@@ -205,15 +205,16 @@ impl VTag {
         }
     }
 
-    /// If there is no ancestor or the classes, or the order, differs from the ancestor:
-    /// - Returns the classes of self separated by spaces.
+    /// Returns the classes of self separated by spaces if and only if
+    /// - the classes, or the order, differs from the ancestor
+    /// - or there is no ancestor and the classes are not empty
```

Comment:
What about the case where this an ancestor and the classes are empty? I think that will cause the empty classes issue as well

Reply:
If the classes are empty and there is an ancestor, there are two cases:
1. The ancestor's classes are empty as well:
    ```rust
    .map(|ancestor| self.classes.ne(&ancestor.classes))
    ```
    will result in false, since both classes are empty.
    Therefore `diff_classes` returns `None`, which will result in no updates to the DOM.

2. The ancestor's classes are not empty (e.g. `class="test"` was set previously):
    ```rust
    .map(|ancestor| self.classes.ne(&ancestor.classes))
    ```
    will result in true, since the length of the classes differs.
    Therefore `diff_classes` returns `Some("")`.
    This results in the DOM-update of `set_attributes("class", "")`, which is needed, because in the DOM, the Element should have a class attribute set, since the ancestor classes are not empty.

Reply:
For 2) is it better to remove the "class" attribute completely?

Reply:
That is a good idea!
I will rework the code to reflect those changes.